src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1,79 +1,1419 @@
+          }
+        })
+        .catch(err => err);
+    } else if (isDraftPlan && planById) {
+      this.setState({ newPlan: planById });
+    }
+
+    // GET PLAN JURISDICTIONS associated with this plan
+    let planJurisdictionIdsToGet: string[] = [];
+    let otherJurisdictionIdsToGet: string[] = [];
+    if (planId && isFinalizedPlan) {
+      const pivotParams = superset.getFormData(500, [
+        { comparator: planId, operator: '==', subject: 'plan_id' },
+      ]);
+      await supersetService(SUPERSET_PLAN_STRUCTURE_PIVOT_SLICE, pivotParams).then(
+        relevantJuridictions => {
+          if (!relevantJuridictions) {
+            return new Promise((resolve, reject) => reject());
+          }
+
+          // define jurisdictions not in the store
+          planJurisdictionIdsToGet = relevantJuridictions
+            .map((j: any) =>
+              !loadedJurisdictionIds.includes(j.jurisdiction_id) ? j.jurisdiction_id : null
+            )
+            .filter((j: string | null) => j && strict.length);
+
+          if (!planJurisdictionIdsToGet.length) {
+            return new Promise(resolve => resolve());
+          }
+
+          // build superset adhoc filter expression
+          let sqlFilterExpression = '';
+          for (let i = 0; i < planJurisdictionIdsToGet.length; i += 1) {
+            const jurId = planJurisdictionIdsToGet[i];
+            if (i) {
+              sqlFilterExpression += ' OR ';
+            }
+            sqlFilterExpression += `jurisdiction_id = '${jurId}'`;
+          }
+
+          // define superset params for
+          const planJurisdictionSupersetParams = superset.getFormData(1000, [
+            { sqlExpression: sqlFilterExpression },
+          ]);
+
+          return supersetService(SUPERSET_JURISDICTIONS_SLICE, planJurisdictionSupersetParams).then(
+            (jurisdictionResults: Jurisdiction[]) =>
+              fetchJurisdictionsActionCreator(jurisdictionResults)
+          );
+        }
+      );
+    }
+
+    // GET REMAINING JURISDICTIONS
+    if (!isFinalizedPlan) {
+      let sqlFilterExpression = '';
+      let idsToUse: string[] = [];
+      if (loadedJurisdictionIds.length || planJurisdictionIdsToGet.length) {
+        otherJurisdictionIdsToGet = allJurIds.filter((j: string) => {
+          return !loadedJurisdictionIds.includes(j) && !planJurisdictionIdsToGet.includes(j);
+        });
+
+        const otherJurisdictionIdsNotToGet = Array.from(
+          new Set([...planJurisdictionIdsToGet, ...loadedJurisdictionIds])
+        );
+
+        const doInclude = otherJurisdictionIdsToGet.length < otherJurisdictionIdsNotToGet.length;
+        idsToUse = doInclude ? otherJurisdictionIdsToGet : otherJurisdictionIdsNotToGet;
+
+        // build query params
+        for (let i = 0; i < idsToUse.length; i += 1) {
+          const jurId = idsToUse[i];
+          if (i) {
+            sqlFilterExpression += doInclude ? ' OR ' : ' AND ';
```

Comment:
I think in many cases using `IN` vs `OR` is usually faster/more efficient.  Do you think we should do that here?  Source: https://stackoverflow.com/a/3074731